cli/app/parse_config.go

```diff
@@ -17,16 +17,24 @@ import (
 	"gopkg.in/yaml.v2"
 )
 
+// TODO: move to config as a methods
+
 // Parse from yaml
-func parseSourceConfig(ctx *cli.Context) (*config.Protocol, *config.Source, error) {
+func parseSourceConfig(ctx *cli.Context) (*config.Node, *config.Source, error) {
 	root := ctx.Path("root")
 
 	if !filepath.IsAbs(root) {
 		return nil, nil, fmt.Errorf("root folder `%s` is not absolute", root)
 	}
 
 	configRoot := root + "/config"
-	configPath := path.Join(configRoot, ctx.String("shape")+".yaml")
+
+	var configPath string
```

Comment:
configPath := ctx.Path("path")

if len(configPath ) < 1{
configPath = path.Join ...
}

Reply:
don't understand

Reply:
No need to do if else